examples/apps/ncp/main.c

```diff
@@ -97,3 +97,24 @@ int main(int argc, char *argv[])
 
     return 0;
 }
+
+
+/*
+ * Provide, if required an "otPlatLog()" function
+ */
+
+#if ((OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_NONE) ||  (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_APP_STREAM))
```

Comment:
I understand and agree.... For `LOG_OUTPUT_NONE` we need a stub/empty implementation of `otLogPlat()`... maybe this can be added in `logging.cpp` so it's common for CLI/NCP builds...

What I am suggesting is that the the default setting in `openthread-core-default-config.h` to be set to  `LOG_OUTPUT_PLATFORM_DEFINED` instead of `OUTPUT_NONE`.... (default is user need to define/provide `otPlatLog` or explicitly set the `LOG_OUTPUT` to some of the other settings...